scripts/linters/js_ts_linter.py

```diff
@@ -238,570 +222,459 @@ def _get_compiled_ts_filepath(self, filepath):
-                        summary_message = (
-                            '%s --> Please add a comment above the @ts-ignore '
-                            'explaining the @ts-ignore at line %s. The format '
-                            'of comment should be -> This throws "...". This '
-                            'needs to be suppressed because ...' % (
-                                file_path, line_number + 1))
-                        python_utils.PRINT(summary_message)
-                        summary_messages.append(summary_message)
-                        python_utils.PRINT('')
+                previous_line_has_ts_ignore = bool(
+                    re.findall(ts_ignore_pattern, line))
 
-                    previous_line_has_comment = bool(
-                        re.findall(comment_pattern, line))
+                if (
+                        previous_line_has_ts_ignore and
+                        not previous_line_has_comment_with_ts_error):
+                    failed = True
+                    summary_message = (
+                        '%s --> Please add a comment above the @ts-ignore '
+                        'explaining the @ts-ignore at line %s. The format '
+                        'of comment should be -> This throws "...". This '
+                        'needs to be suppressed because ...' % (
+                            file_path, line_number + 1))
+                    summary_messages.append(summary_message)
 
-                    previous_line_has_comment_with_ts_error = (
-                        bool(
-                            re.findall(
-                                comment_with_ts_error_pattern, line))
-                        or (
-                            previous_line_has_comment_with_ts_error and
-                            previous_line_has_comment))
+                previous_line_has_comment = bool(
+                    re.findall(comment_pattern, line))
 
-            if failed:
-                summary_message = (
-                    '%s TS ignore check failed' % (
-                        linter_utils.FAILED_MESSAGE_PREFIX))
-            else:
-                summary_message = (
-                    '%s TS ignore check passed' % (
-                        linter_utils.SUCCESS_MESSAGE_PREFIX))
-
-            python_utils.PRINT(summary_message)
-            summary_messages.append(summary_message)
-            python_utils.PRINT('')
+                previous_line_has_comment_with_ts_error = (
+                    bool(
+                        re.findall(
+                            comment_with_ts_error_pattern, line))
+                    or (
+                        previous_line_has_comment_with_ts_error and
+                        previous_line_has_comment))
 
-        return summary_messages
+        return concurrent_task_utils.OutputStream(
+            name, failed, summary_messages, summary_messages)
 
     def _check_ts_expect_error(self):
```

Comment:
Why are these checks not made public unlike other linter files?

Reply:
Because in this linter we have to parse and compile all files before using them in any of the checks, in other words, the checks are dependent, we make these public and test these independently then this will create the problems. Plus, we will migrate all of these to custom eslint checks, therefore I am not seeing any benefit in making these checks public.